File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
string = input()
bomb = input()

stack = []

for s in string:
    
    stack.append(s)
    
    if s == bomb[-1] and ''.join(stack[-len(bomb):]) == bomb:
        del stack[-len(bomb):]
        
if len(stack) == 0:
    print('FRULA')
else:
    print(''.join(stack))

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';

const [string, bomb] = fs.readFileSync(filePath).toString().trim().split('\n');

let stack = [];

for (let s of string) {
  stack.push(s);

  if (
    s === bomb[bomb.length - 1] &&
    stack.slice(-bomb.length).join('') === bomb
  ) {
    stack.splice(-bomb.length, bomb.length);
  }
}

if (stack.length === 0) {
  console.log('FRULA');
} else {
  console.log(stack.join(''));
}
